src/theory/strings/theory_strings.cpp

```diff
@@ -3809,6 +3842,56 @@ void TheoryStrings::registerTerm( Node n, int effort ) {
   }
 }
 
+void TheoryStrings::sendInternalInference(std::vector< Node >& exp, Node conc, const char * c)
+{
+  if( conc.getKind()==AND )
+  {
+    for( const Node& cc : conc )
+    {
+      sendInternalInference( exp, cc, c );
+    }
+    return;
+  }
+  Node lit = conc.getKind()==NOT ? conc[0] : conc;
+  bool pol = conc.getKind()!=NOT;
+  if( lit.getKind()==EQUAL )
+  {
+    for( unsigned i=0; i<2; i++ )
+    {
+      if( !lit[i].isConst() && !hasTerm( lit[i] ) )
+      {
+        // introduces a new non-constant term, do not infer
+        return;
+      }
+    }
+    // does it already hold?
+    if( pol ? areEqual( lit[0], lit[1] ) : areDisequal( lit[0], lit[1] ) )
+    {
+      return;
+    }
+  }
+  else if( lit.isConst() )
+  {
+    if( lit.getConst<bool>() )
```

Comment:
Suggestion: Merge with `else if` condition above.

Reply:
This actually would change the control flow. If lit is const and is false, then we should fall through.

Reply:
I'm stupid.